lisp/ess-r-flymake.el

```diff
@@ -0,0 +1,149 @@
+  (unless (executable-find
+           ;; TODO - check whether lintr package is available
+           "R") (error "Cannot find a suitable R"))
+  ;; If a live process launched in an earlier check was found, that
+  ;; process is killed.  When that process's sentinel eventually runs,
+  ;; it will notice its obsoletion, since it have since reset
+  ;; `ess-r-flymake-proc' to a different value
+  (when (process-live-p ess-r--flymake-proc)
+    (kill-process ess-r--flymake-proc))
+  ;; Save the current buffer, the narrowing restriction, remove any
+  ;; narrowing restriction.
+  (let ((source (current-buffer)))
+    (save-restriction
+      (widen)
+      (setq
+       ess-r--flymake-proc
+       (make-process
+        :name "ess-r-flymake" :noquery t :connection-type 'pipe
+        :buffer (generate-new-buffer " *ess-r-flymake*")
+        :command `("R" "--slave" "--restore" "--no-save" "-e"
+                   ,(eval (concat
+                           "library(lintr);"
+                           ;; commandArgs(TRUE) lets us access
+                           ;; everything after --args as a string:
+                           "try(lint(commandArgs(TRUE)"
+                           ", " "linters = " ess-r-flymake-linters
+                           (when ess-r-flymake-lintr-cache ", cache = TRUE")
+                           "))"))
+                   "--args" ,(eval
+                              ;; text string of the current buffer:
+                              (concat (buffer-substring-no-properties
+                                       (point-min) (point-max)))))
+        :sentinel
+        (lambda (proc _event)
+          (when (eq 'exit (process-status proc))
+            (unwind-protect
+                ;; Only proceed if `proc' is the same as
+                ;; `ess-r--flymake-proc', which indicates that
+                ;; `proc' is not an obsolete process.
+                (if (with-current-buffer source (eq proc ess-r--flymake-proc))
+                    (with-current-buffer (process-buffer proc)
+                      (goto-char (point-min))
+                      ;; Parse the output buffer for diagnostic's
+                      ;; messages and locations, collect them in a list
+                      ;; of objects, and call `report-fn'.
+                      (cl-loop
+                       while (search-forward-regexp
+                              ;; Regex to match the output lint() gives us.
+                              (rx line-start "<text>:"
+                                  ;; row
+                                  (group-n 1 (one-or-more num)) ":"
+                                  ;; column
+                                  (group-n 2 (one-or-more num)) ": "
+                                  ;; type
+                                  (group-n 3 (| "style: " "warning: " "error: "))
+                                  ;; msg
+                                  (group-n 4 (one-or-more not-newline)) "\n")
```

Comment:
Shouldn't it be `line-end` instead of "\n"?

Reply:
Done